mesonbuild/modules/pkgconfig.py

```diff
@@ -78,15 +79,18 @@ def _process_libs(self, libs, public):
                 if public:
                     if not hasattr(obj, 'generated_pc'):
                         obj.generated_pc = self.name
-                    self.add_priv_libs(obj.get_dependencies())
-                    self.add_priv_libs(obj.get_external_deps())
+                    self.add_priv_libs(self.strip_unfound(obj.get_dependencies()))
+                    self.add_priv_libs(self.strip_unfound(obj.get_external_deps()))
```

Comment:
self.add_priv_libs() calls will recurse into _process_libs(). So if you just add the check in dependencies.Dependency case above that should be enough.

I think that's going to break "51 pkgconfig-gen" test actually, because it does:
pc_dep = dependency('libfoo', required : false)
and it assumes that it will add it into requires.private anyway.